symbulate/results.py

```diff
@@ -313,78 +337,100 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
                     a = .02 * (max(x) - min(x))
                     noise = np.random.uniform(low=-a, high=a)
                     x = [i + noise for i in x]
-                # get next color in cycle
-                axes = plt.gca()
-                color = get_next_color(axes)
                 # plot the impulses
-                plt.vlines(x, 0, y, color=color, alpha=alpha, **kwargs)
-                
-                configure_axes(axes, x, y, ylabel = "Relative Frequency" if normalize else "Count")
-            else:
-                raise Exception("Histogram must have type='impulse' or 'bar'.")
+                ax.vlines(x, 0, y, color=color, alpha=alpha, **kwargs)
+                configure_axes(ax, x, y, ylabel="Relative Frequency" if normalize else "Count")
+            if 'rug' in type:
+                if discrete:
+                    self = self + np.random.normal(loc=0, scale=.002 * (max(self) - min(self)), size=len(self))
+                ax.plot(list(self), [0.001]*len(self), '|', linewidth = 5, color='k')
+                if len(type) == 1:
+                    setup_ticks([], [], ax, 'y')
         elif dim == 2:
             x, y = zip(*self)
 
             x_count = count_var(x)
             y_count = count_var(y)
             x_height = x_count.values()
             y_height = y_count.values()
-            
-            discrete_x = sum([(i > 1) for i in x_height]) > .8 * len(x_height)
-            discrete_y = sum([(i > 1) for i in y_height]) > .8 * len(y_height)
+            discrete_x = is_discrete(x_height)
+            discrete_y = is_discrete(y_height)
 
             if type is None:
-                type = "scatter"
-
+                type = ("scatter",)
             if alpha is None:
                 alpha = .5
+            if bins is None:
+                if 'tile' in type:
+                    bins = 10
+                else:
+                    bins = 30
+            if 'tile' in type and (not discrete_x and not discrete_y):
```

Comment:
Wouldn't it make more sense to move this check inside of the `elif 'tile' in type:` block below? 

(I understand why you're doing it -- it's to change the type to scatter for the user. But I would almost prefer that you just raise an Exception and suggest that the user try again with `type='scatter'`.)